### ISSUE TITLE
Can a session get created by a headless server?

### ISSUE THREAD

**Issue by mercmobily** [atom/teletype#265]
Can a session get created by a headless server?
The idea is:

* I initiate a teletype session on a remote server, and get a session key
* Developer A and developer B both get the session key
* Files are changed on the repository on the server

This would be ESPECIALLY cool when the server is actually a live development environment, and changes are being made on the remote code, which will have immediate consequences on the development app.

I realise this isn't an intended use of teletype, but... is it a _possible_ use?

**Comment by Arcanemagus** [comment]
> Can a session get created by a headless server?

You would have to have some way of launching Atom and starting the session in there... alternatively since the protocol backing `teletype` is open and designed so _any_ editor can host it, you can probably put something together with [teletype-client](https://github.com/atom/teletype-client) that runs this for you without Atom involved.

> * Files are changed on the repository on the server

Remote users can't (currently) _save_ the file, so that's a problem that doesn't currently have a solution. You should follow https://github.com/atom/teletype/issues/240 for that.

tl;dr: It's not currently possible.

**Comment by mercmobily** [comment]
I would be interested in implementing it. I had a look at the teletype library, and it looks pretty straightforward. I can wait till #240 is done, and get this going in the meantime. (The only code that baffles me is that `async initialize () { const {ok, body} = await this.restGateway.get('/protocol-version')` but I guess it's time for me to learn this part of ES6)

Can I ask you (or anybody) to give me a quick, general rundown/pointers/gotchas to use teletype-client to create a headless session? I will be happy to write the client's README.md afterwards :D

Thanks!

**Comment by jasonrudolph** [comment]
:wave: Hi @mercmobily: This is such a cool idea. I have to admit that I instantly start daydreaming about the possibilities here. ✨😀

> I realise this isn't an intended use of teletype, but... is it a possible use? ... Can I ask you (or anybody) to give me a quick, general rundown/pointers/gotchas to use teletype-client to create a headless session?

I'm not sure how much work would be involved, but I think it's _probably_ doable. You're right that it's not the *intended* use, and in the near term, we probably don't have the resources to provide support for anyone pursuing this style of use. Studying the code in [atom/teletype](https://github.com/atom/teletype) and the tests in [atom/teletype-client](https://github.com/atom/teletype-client) is likely the best bet for understanding how to interact with teletype-client and for learning how to potentially adapt it for use with a headless session.

We try to treat our issues list as our team's [backlog](https://en.wikipedia.org/wiki/Scrum_(software_development)#Product_backlog), so I'm going to close this issue to keep the backlog clean and focused. I hope that makes sense.

If you continue to pursue this idea, I'd definitely love to hear how it goes! 🍻

**Comment by mercmobily** [comment]
OK thanks a lot
I think I will spend some time as a teletype "user" to get a feel of how things actually work while using it. I will then be able to look at the code and maybe understand it ;D

**Comment by glingy** [comment]
I was looking to do this same thing. Has there been any progress?

**Comment by mercmobily** [comment]
Not on my end. We ended up not using Teletype on our organisation. At the same time, though, #240 is not closed. So... at least in theory, it should be very straightforward to get a client program to create a session and invite specific users.
It would be just so amazing if one of the teletype developers reopened this issue and provided at least a starting point to make this happen...
